types/user.go

```diff
@@ -2,6 +2,15 @@ package types
 
 import "time"
 
+const (
+	ErrEmailNotVerified = UnauthenticatedError("email not verified")
+	ErrInvalidEmail     = InvalidArgumentError("invalid email")
+	ErrInvalidUserName  = InvalidArgumentError("invalid user name")
+	ErrInvalidAvatarURL = InvalidArgumentError("invalid avatar URL")
+	ErrEmailTaken       = ConflictError("email taken")
```

Comment:
Do we want to expose this? It leaks account credentials potentially versus just an "invalid email".

Reply:
I think I agree, lets remove it from the public and fix the API to don't leak this specific error.